sw/host/opentitanlib/src/otp/otp_img.rs

```diff
@@ -2,67 +2,315 @@
+where
+    S: serde::Serializer,
+{
+    serializer.serialize_str("<random>")
+}
+
+impl<'de> Deserialize<'de> for OtpImgValue {
+    fn deserialize<D>(deserializer: D) -> Result<OtpImgValue, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct Visitor;
+
+        impl<'a> de::Visitor<'a> for Visitor {
+            type Value = OtpImgValue;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("an OtpImgValue")
+            }
+
+            fn visit_str<E>(self, val: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Ok(match val {
+                    "<random>" => OtpImgValue::Random,
+                    "true" => OtpImgValue::Bool(true),
+                    "false" => OtpImgValue::Bool(false),
+                    _ => OtpImgValue::Word(
+                        u64::from_str(val)
+                            .map_err(|_| de::Error::invalid_value(Unexpected::Str(val), &self))?,
+                    ),
+                })
+            }
+
+            fn visit_u64<E>(self, val: u64) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Ok(OtpImgValue::Word(val))
+            }
+
+            fn visit_bool<E>(self, val: bool) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Ok(OtpImgValue::Bool(val))
+            }
+
+            fn visit_seq<A>(self, mut val: A) -> Result<Self::Value, A::Error>
+            where
+                A: de::SeqAccess<'a>,
+            {
+                let mut res = Vec::<u32>::new();
+                while let Ok(Some(v)) = val.next_element::<String>() {
+                    res.push(
+                        u32::from_str(&v)
+                            .map_err(|_| de::Error::invalid_value(Unexpected::Str(&v), &self))?,
+                    )
+                }
+                Ok(OtpImgValue::Sequence(res))
+            }
+        }
+        deserializer.deserialize_any(Visitor {})
+    }
+}
+
+#[derive(Annotate, Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct OtpImgItem {
     pub name: String,
-    pub value: DeferredValue,
+    pub value: OtpImgValue,
 }
 
-#[derive(Deserialize, Debug)]
+#[derive(Annotate, Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct OtpImgPartition {
     pub name: String,
     pub items: Option<Vec<OtpImgItem>>,
 }
 
-#[derive(Deserialize, Debug)]
+#[derive(Annotate, Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct OtpImg {
-    pub seed: DecEncoded<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub seed: Option<u64>,
+    // This attribute shouldn't be here, it should be in `OtpImgValue`, but due to a bug in
+    // serde_annotate this is needed to get values serialized in hex.
+    #[annotate(format = hex)]
```

Comment:
do we still need this? Asking because of the recent update to serde_annotate

Reply:
I've filed an issue for this, see https://github.com/cfrantz/serde-annotate/issues/5